api/envoy/service/ext_proc/v3/external_processor.proto

```diff
@@ -170,6 +171,17 @@ message ProcessingResponse {
   // may use this to intelligently control how requests are processed
   // based on the headers and other metadata that they see.
   envoy.extensions.filters.http.ext_proc.v3.ProcessingMode mode_override = 9;
+
+  // When ext_proc server receives a request message, in case it needs more
+  // time to process the message, it sends back a ProcessingResponse message
+  // with a new timeout value. When Envoy receives this response message,
+  // it ignores other fields in the response, just stop the original timer,
+  // which has the timeout value specified in
+  // :ref:`message_timeout
+  // <envoy_v3_api_field_extensions.filters.http.ext_proc.v3.ExternalProcessor.message_timeout>`
+  // and start a new timer with this ``new_timeout`` value and keep the Envoy ext_proc
```

Comment:
Are there any security implications involved in trusting the ext_proc server to specify an arbitrary timeout here?

Also, I assume that the data plane request will still be bound by its configured timeout, regardless of what timeout is used for the ext_proc message, right?

Reply:
For the 1st comment: the new timeout is bound to be >=1ms and <=10s given the info I got that WASM module cold start could take up to a few seconds.  If the specified new timeout is out of this range, then this new timeout update API will be ignored by Envoy. 

For the 2nd comment: no, once Envoy receive this new timeout API,  it stops the original time and  start a new timer with the new timeout value.

Reply:
Okay, having some hard-coded bounds here is a good idea.

The data plane request timeout I was asking about is the one controlled by the [`max_stream_duration`](https://github.com/envoyproxy/envoy/blob/57be3189ffa3372b34e9480d1f02b2d165e49077/api/envoy/config/route/v3/route_components.proto#L1363) field.  It sounds like this does not affect that timeout, which it shouldn't.